crates/y-sweet-worker/src/s3_store.rs

```diff
@@ -17,19 +20,18 @@ pub struct S3Store {
 }
 
 impl S3Store {
-    pub fn new(
-        region: String,
-        bucket_name: String,
-        prefix: Option<String>,
-        aws_access_key_id: String,
-        aws_secret: String,
-    ) -> Self {
-        let credentials = Credentials::new(aws_access_key_id, aws_secret);
-        let endpoint = format!("https://s3.dualstack.{}.amazonaws.com", region)
-            .parse()
-            .expect("endpoint is a valid Url");
-        let path_style = rusty_s3::UrlStyle::VirtualHost;
-        let bucket = Bucket::new(endpoint, path_style, bucket_name, region)
+    pub fn new(config: S3Config) -> Self {
+        let credentials = Credentials::new(config.key, config.secret);
+        let endpoint: Url = config.endpoint.parse().expect("endpoint is a valid url");
+        let path_style =
+            // if endpoint is localhost then bucket url must be of forme http://localhost:<port>/<bucket>
+            // instead of <method>:://<bucket>.<endpoint>
+            if endpoint.host_str().expect("endpoint Url should have host") == "localhost" {
```

Comment:
Not a big deal since we will only use this for testing, but fwiw this will not work for 127.0.0.1. I sometimes use 127.0.0.1 over `localhost` because it unambiguously refers to IPv4 localhost and not IPv6 localhost.